rootston/seat.c

```diff
@@ -714,6 +724,17 @@ void roots_seat_set_focus(struct roots_seat *seat, struct roots_view *view) {
 		wl_list_insert(&seat->input->server->desktop->views, &view->link);
 	}
 
+	struct roots_desktop *desktop = view->desktop;
+	struct roots_output *output;
+	struct wlr_box box;
+	view_get_box(view, &box);
+	wl_list_for_each(output, &desktop->outputs, link) {
+		if (wlr_output_layout_intersects(desktop->layout,
+				output->wlr_output, &box)) {
```

Comment:
I don't think this works with xwayland popups for fullscreened xwayland views.

Reply:
Works better than firefox going completly black for me on the current master if it's right clicked while fullscreen...

Got a likely fix, but can't exactly test with this behaviour